src/js/components/Select/__tests__/Select-test.js

```diff
@@ -1709,6 +1715,208 @@ describe('Select', () => {
+          value="one"
+        />
+      </Grommet>,
+    );
+
+    const select = document.querySelector('#test-select');
+
+    downKey(select);
+
+    await waitFor(() =>
+      expect(
+        document.querySelector('#test-select__select-drop'),
+      ).toBeInTheDocument(),
+    );
+
+    const search = screen.getByPlaceholderText('test search');
+    const clear = screen.getByLabelText(/Clear selection/);
+
+    await waitFor(() => expect(search).toHaveFocus());
+
+    const options = await screen.findAllByRole('option');
+
+    downKey();
+    expect(clear).toHaveFocus();
+    downKey();
+    expect(options[0]).toHaveFocus();
+    downKey();
+    expect(options[1]).toHaveFocus();
+    downKey();
+    // Focus should stay on the last selectable option
+    expect(options[1]).toHaveFocus();
+
+    upKey();
+    expect(options[0]).toHaveFocus();
+    upKey();
+    expect(clear).toHaveFocus();
+    upKey();
+    expect(search).toHaveFocus();
+    upKey();
+    // Focus should stay on the search input
+    expect(search).toHaveFocus();
+  });
+
+  test('allows keyboard navigation with search and bottom clear', async () => {
+    // Neither user.type() nor user.keyboard() trigger arrow keys as needed
+    const downKey = (el) =>
+      fireEvent.keyDown(el ?? document.activeElement, {
+        key: 'ArrowDown',
+        keyCode: 40,
+        which: 40,
+      });
+    const upKey = (el) =>
+      fireEvent.keyDown(el ?? document.activeElement, {
+        key: 'ArrowUp',
+        keyCode: 38,
+        which: 38,
+      });
+
+    const onSearch = jest.fn();
+
+    render(
+      <Grommet>
+        <Select
+          clear={{ position: 'bottom' }}
+          id="test-select"
+          options={['one', 'two']}
+          onSearch={onSearch}
+          searchPlaceholder="test search"
+          // Preselect option to make clear immediately available
+          value="one"
+        />
+      </Grommet>,
+    );
+
+    const select = document.querySelector('#test-select');
+
+    downKey(select);
+
+    await waitFor(() =>
+      expect(
+        document.querySelector('#test-select__select-drop'),
+      ).toBeInTheDocument(),
+    );
+
+    const search = screen.getByPlaceholderText('test search');
+    const clear = screen.getByLabelText(/Clear selection/);
+
+    await waitFor(() => expect(search).toHaveFocus());
+
+    const options = await screen.findAllByRole('option');
+
+    downKey();
+    expect(options[0]).toHaveFocus();
+    downKey();
+    expect(options[1]).toHaveFocus();
+    downKey();
+    expect(clear).toHaveFocus();
+    // Focus should stay on clear
+    expect(clear).toHaveFocus();
+
+    upKey();
+    expect(options[1]).toHaveFocus();
+    upKey();
+    expect(options[0]).toHaveFocus();
+    upKey();
+    expect(search).toHaveFocus();
+    upKey();
+    // Focus should stay on the search input
+    expect(search).toHaveFocus();
+  });
+
+  test('does not scroll on search or bottom clear focus', async () => {
```

Comment:
This test wasn't part of the original changeset and I've added it to cover the case concerning the undesirable scrolling related to the clear button focus.